app.js

```diff
@@ -69,108 +66,89 @@ class ChallengeCard {
-    labelsElement.classList.add("labels");
-    labels.forEach((label) => {
-      const labelElement = document.createElement("span");
-      labelElement.textContent = label;
-      labelsElement.appendChild(labelElement);
-    });
+    detailsContainer.appendChild(typeOfRoom);
 
     const imageElement = document.createElement("img");
-    imageElement.src = image;
-    imageElement.alt = title;
+    imageElement.src = this.data.image;
+    imageElement.alt = this.data.title;
     imageElement.classList.add("rooms__img");
 
+    const bookButton = document.createElement("button");
+    bookButton.textContent = "Book this room";
+    bookButton.classList.add("rooms__button");
+
     card.appendChild(imageElement);
     card.appendChild(titleElement);
-    card.appendChild(detailsContainer);
     card.appendChild(descriptionElement);
-    card.appendChild(labelsElement);
 
     return card;
   }
+}
 
-  getCardElement() {
-    return this.cardElement;
+// FETCH API
+class APIadapter {
+  async fetchAllChallenges() {
+    const url = "https://lernia-sjj-assignments.vercel.app/api/challenges";
+    const response = await fetch(url);
+    const data = await response.json();
+    console.log("Fetched Challenges:", data);
+
+    return data.challenges.map(
+      (challengeData) => new ChallengeCard(challengeData)
+    );
   }
+}
 
-  static async fetchData(apiUrl) {
+class ViewAll {
+  async render(container) {
     try {
-      const response = await fetch(apiUrl);
-      const data = await response.json();
-      console.log(data);
-      return data;
-    } catch (error) {
-      console.error("Kunde inte ladda korten :(", error);
-      throw error;
-    }
-  }
+      const api = new APIadapter();
+      const challenges = await api.fetchAllChallenges();
+      console.log("All Challenges:", challenges);
 
-  static async createCards(apiUrl) {
-    try {
-      const apiData = await ChallengeCard.fetchData(apiUrl);
-      return apiData.challenges.map(
-        (challenge) => new ChallengeCard(challenge)
-      );
+      challenges.forEach((challenge) => {
+        const element = challenge.render();
+        container.appendChild(element);
+      });
     } catch (error) {
-      console.error("Kunde inte ladda korten :(", error);
-      throw error;
+      console.error("Error rendering all challenges:", error);
     }
   }
 }
-/*
-class ViewAll {
+
+class ViewTopThree {
   async render(container) {
-    const api = new APIadapter();
-    const challenges = await api.fetchChallenges();
-    for (let i = 0; i < challenges.length; i++) {
-      const challenge = challenges[i];
-      const element = challenge.render();
-      container.append(element);
+    try {
+      const api = new APIadapter();
+      const challenges = await api.fetchAllChallenges();
+
+      const sorted = challenges.sort((a, b) => b.data.rating - a.data.rating);
+      const topThree = sorted.slice(0, 3);
+
+      console.log("Top Three Challenges:", topThree);
```

Comment:
Generally, it's good practice to take away all `console.log` before committing. It's a great tool for developing and debugging, and understanding what happens in your code. But if you leave lots of them in the code, it actually makes debugging (other parts of the code) more complicated for future you and your team mates.